src/annotator/highlighter.ts

```diff
@@ -12,6 +12,57 @@ type HighlightProps = {
 
 export type HighlightElement = HTMLElement & HighlightProps;
 
+/**
+ * Return the closest generation of HighlightElements to `element`.
+ *
+ * If `element` is itself a HighlightElement, return immediate children that
+ * are also HighlightElements.
+ *
+ * Otherwise, return all HighlightElements that have no parent HighlightElement,
+ * i.e. the outermost highlights within `element`.
+ */
+export function getHighlights(element: Element) {
+  let highlights;
+  if (element.tagName === 'HYPOTHESIS-HIGHLIGHT') {
+    highlights = Array.from(element.children).filter(
+      el => el.tagName === 'HYPOTHESIS-HIGHLIGHT'
+    );
+  } else {
+    highlights = Array.from(
+      element.getElementsByTagName('HYPOTHESIS-HIGHLIGHT')
+    ).filter(
+      highlight => highlight.parentElement?.tagName !== 'HYPOTHESIS-HIGHLIGHT'
+    );
+  }
+  return highlights as HighlightElement[];
+}
+
+/**
+ * Get all of the SVG highlights within `root`, grouped by layer. A PDF
+ * document may have multiple layers of SVG highlights, typically one per page.
+ *
+ * @return a Map of layer Elements to all SVG highlights within that
+ *   layer Element
+ */
+export function getSVGHighlights(
+  root?: Element
```

Comment:
You could also use a default argument here to avoid the `root ?? document` below.